packages/webpack/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for @webpack-blocks/webpack 1.0.0
+// Project: webpack-blocks
+// Definitions by: Stephan Schneider <https://github.com/zcei>
+
+import {
+  createConfig,
+  group,
+  env,
+  match,
+  WebpackBlock,
+  WebpackBlockUpdater
+} from '@webpack-blocks/core'
+import { Configuration, Plugin, Entry, Options, Resolve, Output } from 'webpack'
+export { createConfig, group, env, match }
+
+export function addPlugins(plugins: Plugin[]): WebpackBlock
+export function customConfig(configSnippet: Configuration): WebpackBlockUpdater
```

Comment:
Actually returns a `WebpackBlock`

Reply:
I think we have to agree upon some terminology. The term `block`, for instance, has always been a bit fuzzy. Providing explicit types is probably the ideal time to define it properly.

Reply:
Initially I was thinking that a `block` is a self-contained, configurable portion of a Webpack Configuration - so actually `entryPoint` is the block, not `entryPoint()`, which could be a `BlockInstance` or similar.

Then a `block` is everything that has the signature `(...args: any[]): BlockInstance`, where `BlockInstance = (current) WebpackBlock`

Reply:
Fair enough. I would follow your argumentation here, @zcei. `BlockInstance` is not very catchy, but should do for now.